controllers/awscluster_controller.go

```diff
@@ -87,49 +119,83 @@ func (r *AWSClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, err
 	}
 
+	cniSecurityGroupID := awsCluster.Status.Network.SecurityGroups[key.CNINodeSecurityGroupName].ID
+
 	var cniService *cni.CNIService
-	{
+	// config for the CNI service
+	config := cni.CNIConfig{
+		AWSSession:         awsClientSession,
+		ClusterName:        clusterName,
+		CNISecurityGroupID: cniSecurityGroupID,
+		CtrlClient:         nil,              // we only need wc k8s client for resource creation, we dont need it for deletion, when cluster is being deleted it might not be avaiable
```

Comment:
Why do we set it explicitly to `nil`? I think if it were me I'd just leave it out. (Not saying that is better, just curios)

Reply:
i mostly wanted to put the comment there to explain why we do not set it here but later

I can remove it

Reply:
Ah no that makes perfect sense! No need to remove it. :)